### PR TITLE
Create export_for_inference API

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #135269
* #135080
* #134801

Differential Revision: [D62265487](https://our.internmc.facebook.com/intern/diff/D62265487/)

### CODE REVIEW

**Review by avikchaudhuri — APPROVED** [review]
lgtm, fix lint

File: torch/export/__init__.py [hunk 1]
```diff
@@ -155,6 +157,91 @@ def export_for_training(
     )
 
 
+def export_for_inference(
+    mod: torch.nn.Module,
```
Comment (zhxchen17) [inline torch/export/__init__.py:161]: @tugsbayasgalan I think we can add support for non-nn.Module as well. A lot of people only need to export a free function and if it's ONLY for inference, we can assume they probably don't depend on some node meta that is only available under training.

If you don't have time, we can consider opening a task and leave a TODO.

**Comment by bhack** [comment]
What is the expected use of `export_for_inference` vs `export`?

**Comment by avikchaudhuri** [comment]
@bhack export by default generates high level IR (non-functional full aten) whereas with `run_decompositions` you'd get low level IR (functional core aten).

**Comment by bhack** [comment]
@avikchaudhuri So what we need to use e.g. for aoti compile or other use cases?

**Pull request closed without merging**